R/loadFromMetaphlan.R

```diff
@@ -392,3 +394,40 @@ loadFromMetaphlan <- function(
     }
     return(data)
 }
+.set_taxonomy_ranks <- function(tse, ranks = NULL, set.ranks = FALSE,...) {
+    # Input checks
+    if (!is.logical(set.ranks) || length(set.ranks) != 1) {
+        stop("'set.ranks' must be a logical scalar.", call. = FALSE)
+    }
+    
+    if (!is.null(ranks) && (!is.character(ranks) || length(ranks) == 0)) {
+        stop("'ranks' must be a non-empty character vector if provided.", call. = FALSE)
+    }
+    
+    # Attempt to retrieve ranks if NULL
+    if (is.null(ranks)) {
+        if (!is.null(rowData(tse))) {
+            # Filter column names by those that are character type
+            char_cols <- sapply(rowData(tse), is.character)
+            ranks <- colnames(rowData(tse))[char_cols]
+            
+            if (length(ranks) == 0 && !set.ranks) {
+                stop("No character-type column names in 'rowData(tse)' to infer ranks and 'set.ranks' is TRUE.", call. = FALSE)
+            }
+        } else {
+            ranks <- character(0)
+            #stop("rowData of the 'tse' object does not exist.", call. = FALSE)
+        }
+        
+    }
+    new_ranks_not_in_existing <- setdiff(ranks, TAXONOMY_RANKS)
```

Comment:
This check if rowData includes ranks that are not included in TAXONOMY_RANKS. Below, the message advices to use setTaxonomyRanks. The idea of this function is to add those ranks to TAXONOMY_RANKS that are not included in it by default. For example, strain is usual in Metaphlan files.

So, instead of giving just warning, the function should change the ranks